components/script/dom/url.rs

```diff
@@ -24,19 +26,24 @@ pub struct URL {
 
     // https://url.spec.whatwg.org/#concept-urlutils-get-the-base
     base: Option<Url>,
+
+    // https://url.spec.whatwg.org/#dom-url-searchparams
+    query: JS<URLSearchParams>,
```

Comment:
Technically we can't; a quick test in Firefox shows that mutating the object returned on `url.searchParams` will be reflected when accessed again. The spec does mention that a `URL` object has an associated `URLSearchParams` object, so this is correct I guess.

cc @Ms2ger does this particularly matter?